quinn-proto/src/connection/streams.rs

```diff
@@ -1034,6 +1042,39 @@ impl Recv {
         }
     }
 
+    /// Returns the window that should be advertised in a MAX_STREAM_DATA frame
+    ///
+    /// If there is no window update necessary at the moment, this returns `None`.
+    ///
+    /// If the `peek` parameter is set to `false`, the method will assume the
+    /// window update will actually be enqueued for transmission by the caller,
+    /// and track the transmission. Therefore a immediate follow-up call to this
+    /// method would return `None`.
+    ///
+    /// If `peek` is set to `false`, no transmission of the outgoing update would
+    /// be recorded.
+    fn window_update(&mut self, stream_receive_window: u64, peek: bool) -> Option<u64> {
```

Comment:
I think callsites would be a bit more readable if `peek` was removed in favor of a separate call (or direct assignment) that updates the counter.

Reply:
I had the separate assignment initially. But I removed it after djc's comments to make the `Streams` struct more self-contained. A separate method would be possible

Reply:
I definitely think a separate method is clearer. @djc, thoughts?

Reply:
This was now kind of necessary to properly handle the retransmits anyway